### PR TITLE
fix(Components): tests with @talend/scripts

### PR DESCRIPTION
**Pull request by jsomsanith-tlnd**
WARNING: this is to merge to jsomsanith/chore/talend_scripts because of eslint rules at root. It conflicts with talend/scripts config and breaks the lint. Let's move all packages to @talend/scripts before the merge to master.

**What is the problem this PR is trying to solve?**
Components tests must be adapted to talend/scripts jest/enzyme/mock versions

**What is the chosen solution to this problem?**
Fix a part of them

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/ActionBar/ActionBar.component.js [hunk 1]
```diff
@@ -141,7 +140,7 @@ function SwitchActions({ actions, left, right, center, getComponent, components
 	);
 }
 SwitchActions.propTypes = {
-	actions: PropTypes.arrayOf(PropTypes.shape(actionsShape)).isRequired,
+	actions: PropTypes.arrayOf(PropTypes.shape(actionsShape)),
```
Comment (jsomsanith-tlnd) [inline packages/components/src/ActionBar/ActionBar.component.js:143]: I removed the required because of console error/warning. This has a default props.

File: packages/components/src/ActionBar/ActionBar.component.js [hunk 2]
```diff
@@ -223,24 +222,22 @@ export function ActionBarComponent(props) {
...
-export default TranslatedActionBar;
+ActionBar.displayName = 'ActionBar';
+
+ActionBar.DISPLAY_MODES = DISPLAY_MODES;
+ActionBar.Count = Count;
+ActionBar.SwitchActions = SwitchActions;
+ActionBar.getActionsToRender = getActionsToRender;
+ActionBar.Content = Content;
+ActionBar.getContentClassName = getContentClassName;
+export default ActionBar;
```
Comment (jsomsanith-tlnd) [inline packages/components/src/ActionBar/ActionBar.component.js:243]: Removed the HOC, where the ActionBar pass the `t()` to only `Count`. Use directly i18next in `Count` via hook. That will simplify ActionBar  selections in tests

File: packages/components/src/DateTimePickers/LegacyDateTimePickers/views/HeaderTitle/HeaderTitle.test.js [hunk 3]
```diff
@@ -19,7 +19,12 @@ describe('HeaderTitle', () => {
 	it('should render a button', () => {
 		// When
 		const wrapper = shallow(
-			<HeaderTitle monthIndex={8} year={2012} button={{ whateverButtonProp: 'whateverValue' }} />,
+			<HeaderTitle
+				monthIndex={8}
+				year={2012}
+				button={{ whateverButtonProp: 'whateverValue' }}
+				onSelectYear={jest.fn()}
```
Comment (jsomsanith-tlnd) [inline packages/components/src/DateTimePickers/LegacyDateTimePickers/views/HeaderTitle/HeaderTitle.test.js:26]: Added required `onSelectYear ` in all this files tests to remove warnings

File: packages/components/src/Drawer/__snapshots__/Drawer.test.js.snap [hunk 4]
```diff
@@ -302,9 +302,6 @@ exports[`Drawer should render with tabs specific actions by tab with selectedTab
   }
   className="tc-drawer-actionbar theme-tc-drawer-actionbar"
   components={Object {}}
-  i18n={Object {}}
-  t={[Function]}
-  tReady={true}
```
Comment (jsomsanith-tlnd) [inline packages/components/src/Drawer/__snapshots__/Drawer.test.js.snap:307]: Due to ActionBar i18n HOC removal

File: packages/components/src/Drawer/__snapshots__/Drawer.test.js.snap [hunk 5]
```diff
@@ -341,53 +336,46 @@ exports[`Drawer should render with tabs specific actions by tab with selectedTab
...
                 id="view-center-tab-2"
-                inProgress={false}
-                label="ActionCenter-tab-2"
-                loading={false}
-                t={[Function]}
-                tReady={true}
-                tooltipPlacement="top"
+                onClick={[Function]}
+                onMouseDown={[Function]}
+                role={null}
```
Comment (jsomsanith-tlnd) [inline packages/components/src/Drawer/__snapshots__/Drawer.test.js.snap:360]: This change is here because of the talend/script i18n HOC mock. It returns the original component instead of wrapping it.

Before 
```
<withI18nextTranslation(ActionButton)>
    <ActionButton>
        ... // children
    </ActionButton>
</withI18nextTranslation(ActionButton)>
```

After
```
<withI18nextTranslation(ActionButton)>
    ... // children
</withI18nextTranslation(ActionButton)>
```

File: packages/components/src/FilterBar/FilterBar.test.js [hunk 6]
```diff
@@ -131,14 +131,17 @@ describe('FilterBar', () => {
 	it('should call onToggle on ENTER keydown', () => {
 		// given
 		const props = { ...defaultProps };
-		const filterInstance = mount(<FilterBarComponent {...props} id={'filter'} />);
+		const filterInstance = mount(<FilterBarComponent {...props} id="filter" />, {
+			attachTo: document.body,
```
Comment (jsomsanith-tlnd) [inline packages/components/src/FilterBar/FilterBar.test.js:135]: We select the activeElement. With talend/scripts version of jest, we need to attach the component to the jsdom.

File: packages/components/src/FilterBar/FilterBar.test.js [hunk 7]
```diff
@@ -131,14 +131,17 @@ describe('FilterBar', () => {
...
+		});
 		expect(document.activeElement.id).toBe('filter-input');
 
 		// when
 		filterInstance.find('input').simulate('keydown', { keyCode: 13 });
 
 		// then
 		expect(document.activeElement.id).toBe('');
+		filterInstance.detach();
```
Comment (jsomsanith-tlnd) [inline packages/components/src/FilterBar/FilterBar.test.js:144]: And to detach it to not pollute the other tests.

File: packages/components/src/HeaderBar/HeaderBar.test.js [hunk 8]
```diff
@@ -4,7 +4,7 @@ import HeaderBarComponent from './HeaderBar.component';
 
 describe('HeaderBar', () => {
 	it('should render', () => {
-		const wrapper = mount(<HeaderBarComponent.WrappedComponent />);
+		const wrapper = mount(<HeaderBarComponent />);
```
Comment (jsomsanith-tlnd) [inline packages/components/src/HeaderBar/HeaderBar.test.js:7]: Talend/scripts i18n HOC mock returns the original component. No WrappedComponent anymore

File: packages/components/src/HeaderBar/HeaderBar.test.js [hunk 9]
```diff
@@ -115,20 +114,17 @@ describe('HeaderBar', () => {
...
 		expect(products.onSelect).toHaveBeenCalled();
 	});
 
 	it('should render intercom', () => {
 		// when
 		const wrapper = shallow(
-			<HeaderBarComponent.WrappedComponent
-				intercom={{ id: 'my-intercom', config: { app_id: 'e19c98d' } }}
+			<HeaderBarComponent
+				intercom={{ id: 'my-intercom', config: { app_id: 'e19c98d', email: 'lol@lol.com' } }}
```
Comment (jsomsanith-tlnd) [inline packages/components/src/HeaderBar/HeaderBar.test.js:127]: Missing required prop

File: packages/components/src/HeaderBar/HeaderBar.test.js [hunk 10]
```diff
@@ -138,10 +134,12 @@ describe('HeaderBar', () => {
...
-		expect(intercomTrigger.props()).toEqual({
-			className: 'btn btn-link',
-			id: 'my-intercom',
-			config: { app_id: 'e19c98d', vertical_padding: 70 },
+		expect(intercomTrigger.prop('className')).toEqual('btn btn-link');
+		expect(intercomTrigger.prop('id')).toEqual('my-intercom');
+		expect(intercomTrigger.prop('config')).toEqual({
+			app_id: 'e19c98d',
+			email: 'lol@lol.com',
+			vertical_padding: 70,
```
Comment (jsomsanith-tlnd) [inline packages/components/src/HeaderBar/HeaderBar.test.js:142]: I changed this assertion to multiple because `props()` now returns the element default props too. So let's only check the info we wanted too.
